### DISCUSSION TITLE
How to retrieve initialState passed to a View?

### DISCUSSION THREAD

**Discussion by svengato** [GMOD/jbrowse-components#3942] [Q&A]
My plugin's `configure()` method adds a view, which should have an initialState {x, y}.
```
session.addView('myView', { x: 42, y: 3.14159, })
```

How do I retrieve the initial values {x, y} in the view code (MyView.tsx)?
```
export default function ReactComponent() {
  const [x, setX] = useState(initialState.x) // initialState is undefined
  const [y, setY] = useState(initialState.y)
}
```

**Comment by garrettjstevens** [comment]
Those values are passed to the creation of the view state, so assuming you have defined `x` and `y` in your view model, those will be available as part of the `model` prop passed to the view component.

```js
function ReactComponent(props) {
  const {x, y} = props.model
}
```

**Comment by svengato** [comment]
No, they do not appear there. I am checking to make sure I did not leave out anything important.

**Reply by garrettjstevens** [reply]
Can you show me how you have `x` and `y` defined on your view model?

**Comment by svengato** [comment]
I had not defined them in the model. This works,
```
const stateModel = types
  .model({
    id: ElementId,
    type: types.literal('MyView'),
    x: types.number,
    y: types.number,
  })
```
Thanks!
